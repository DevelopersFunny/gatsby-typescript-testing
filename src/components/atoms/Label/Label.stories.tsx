import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import { Label } from '.'

export default {
  title: 'Atoms/Label',
  component: Label,
  parameters: {
    fileName: __filename,
  },
  argTypes: {
    backgroundColor: { control: 'color' },
    textColor: { control: 'color' },
  },
} as ComponentMeta<typeof Label>

const Template: ComponentStory<typeof Label> = args => <Label {...args} />

export const LabelName = Template.bind({})
LabelName.args = {
  children: 'Label',
}
