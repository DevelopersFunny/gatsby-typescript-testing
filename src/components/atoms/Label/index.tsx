import React, { ReactNode } from 'react'
import tw, { styled } from 'twin.macro'

export interface ILabelProps {
  className?: string
  backgroundColor?: string
  textColor?: string
  htmlFor?: string
  children?: ReactNode
}

const StyledLabel = styled.label([
  tw`
    block
    text-sm
    font-medium
    text-gray-700
  `,
])

export function Label({
  backgroundColor,
  className,
  htmlFor,
  children,
  textColor,
}: ILabelProps) {
  return (
    <StyledLabel
      htmlFor={htmlFor}
      className={className}
      style={{ backgroundColor, color: textColor }}
    >
      {children}
    </StyledLabel>
  )
}
