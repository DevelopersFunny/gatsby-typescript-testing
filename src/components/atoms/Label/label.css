.labelName {
  @apply block text-sm font-medium text-slate-700;
}
