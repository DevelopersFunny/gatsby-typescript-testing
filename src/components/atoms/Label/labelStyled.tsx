import tw, { styled } from 'twin.macro'

export const StyledLabel = styled.label([
  tw`
    block
    text-sm
    font-medium
    text-gray-700
  `,
])
