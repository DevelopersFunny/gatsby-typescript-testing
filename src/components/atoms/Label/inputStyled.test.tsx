import React from 'react'
import * as stories from './Label.stories'
import { composeStories } from '@storybook/testing-react'
import { render, screen } from '@testing-library/react'
import { createSerializer, matchers } from '@emotion/jest'

/**
 * Emotion matchers and snapshots
 */
expect.extend(matchers)
expect.addSnapshotSerializer(createSerializer())

/**
 * Stories render
 */
const { LabelName } = composeStories(stories)

describe('Input render with props', () => {
  it('should render input', () => {
    render(<LabelName>Label</LabelName>)
    const labelElement = screen.getByText('Label')
    expect(labelElement).toMatchSnapshot()
  })
})
