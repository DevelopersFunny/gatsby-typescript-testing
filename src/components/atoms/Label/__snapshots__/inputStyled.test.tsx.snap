// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render with props should render input 1`] = `
.emotion-0 {
  display: block;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 500;
  --tw-text-opacity: 1;
  color: rgba(55, 65, 81, var(--tw-text-opacity));
}

.emotion-0 {
  display: block;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 500;
  --tw-text-opacity: 1;
  color: rgba(55, 65, 81, var(--tw-text-opacity));
}

<label
  class="emotion-0"
>
  Label
</label>
`;
